vector/src/main/java/im/vector/app/features/home/room/detail/composer/MessageComposerViewModel.kt

```diff
@@ -741,14 +745,32 @@ class MessageComposerViewModel @AssistedInject constructor(
         voiceMessageHelper.startOrPauseRecordingPlayback()
     }
 
-    private fun handleEndAllVoiceActions(deleteRecord: Boolean) {
-        voiceMessageHelper.stopAllVoiceActions(deleteRecord)
+    private fun handleEndAllVoiceActions(deleteRecord: Boolean): MultiPickerAudioType? {
+        return voiceMessageHelper.stopAllVoiceActions(deleteRecord)
+    }
+
+    private fun handleInitializeVoiceRecorder(attachmentData: ContentAttachmentData) {
+        voiceMessageHelper.initializeRecorder(room.roomId, attachmentData)
+        setState { copy(voiceRecordingUiState = VoiceMessageRecorderView.RecordingUiState.Draft) }
     }
 
     private fun handlePauseRecordingVoiceMessage() {
         voiceMessageHelper.pauseRecording()
     }
 
+    private fun handleEntersBackground(composerText: String) {
+        withState {
+            if (it.isVoiceRecording) {
+                handleEndAllVoiceActions(deleteRecord = false)?.toContentAttachmentData()?.let { voiceDraft ->
+                    handleSaveDraft(draft = voiceDraft.toJsonString(), messageType = MessageType.MSGTYPE_AUDIO)
+                }
+                setState { it.copy(voiceRecordingUiState = VoiceMessageRecorderView.RecordingUiState.None) }
```

Comment:
Is it worth (re-)setting the state when entering Background? What about screen rotation?

Reply:
will check

Reply:
this state update isn't needed, will remove and for rotation the recording stops and the playback composer is displayed until sent or deleted

Reply:
Nice to have: the recording should not stop on screen rotation... (maybe handle that in another PR)

Reply:
Not stopping the recording will probably mean moving the recording/playback to a service (it would also enable recording/playing whilst the screen is off or in other screens), would need its own ticket/issue

The rotation is already a bit funny in develop~

| ROTATION |
| --- | 
![vm-rotation](https://user-images.githubusercontent.com/1848238/143029686-aa3b25a5-737c-40ac-8c29-8a0b518a15fb.gif)